ext/backtracie_native_extension/backtracie_frames.c

```diff
@@ -554,23 +561,16 @@ static void mod_to_s(VALUE klass, strbuilder_t *strout) {
   strbuilder_append_value(strout, klass_name);
 }
 
-static void method_qualifier(const raw_location *loc, strbuilder_t *strout) {
-  rb_callable_method_entry_t *cme =
-      (rb_callable_method_entry_t *)loc->callable_method_entry;
-  VALUE defined_class = cme ? cme->defined_class : Qnil;
-  VALUE class_of_defined_class =
-      RTEST(defined_class) ? rb_class_of(defined_class) : Qnil;
-  VALUE self = loc->self_is_real_self ? loc->self_or_self_class : Qundef;
-  VALUE self_class = loc->self_is_real_self
-                         ? rb_class_of(loc->self_or_self_class)
-                         : loc->self_or_self_class;
-  VALUE method_target = RTEST(defined_class) ? defined_class : self_class;
-
+static void method_qualifier_impl(VALUE defined_class,
+                                  VALUE class_of_defined_class, VALUE self,
+                                  VALUE self_class, VALUE method_target,
+                                  strbuilder_t *strout) {
```

Comment:
This method no longer needs `defined_class` nor `self_class`, since they were only needed to compute derived stuff that is now directly handed over by the callers 😄